src/mgr/MgrPyModule.h

```diff
@@ -46,8 +47,8 @@ class MgrPyModule
 private:
   const std::string module_name;
   PyObject *pClassInstance;
-  PyThreadState *pMainThreadState;
-  PyThreadState *pMyThreadState = nullptr;
+  SafeThreadState pMainThreadState;
```

Comment:
nit, they are not pointers anymore. also, i think we might want to stick with the naming convention used by boost, i.e. `not_camel_case` instead of the `camelCase`.

Reply:
For the moment I would like to keep the name the same to make merging with my other branches easier